Tons of bug fixes and updates for pyright

naddeoa opened the pull request:
Started using pyright and it immediately exposed tons of issues that mypy wasn't catching. I updated the actor stuff to reflect it since there's still hope that I can fix all of the errors in the container. There are a bunch of actual errors though, not just type issues.

Commit:
Tons of bug fixes and updates for pyright

Started using pyright and it immediately exposed tons of issues that
mypy wasn't catching. I updated the actor stuff to reflect it since
there's still hope that I can fix all of the errors in the container.
There are a bunch of actual errors though, not just type issues.

richard-rogers commented on python/whylogs/api/logger/experimental/logger/actor/thread_rolling_logger.py:
```diff
@@ -111,11 +116,9 @@ def to_views(self) -> List[DatasetProfileView]:
+        else:
             return [self._target.view()]
 
-        raise Exception("Unknown profile type")
```
Raising a plain `Exception` is probably  not optimal, but do we not want to throw  here?

naddeoa commented on python/whylogs/api/writer/whylabs.py:
```diff
@@ -115,8 +115,8 @@ class WhyLabsWriter(Writer):
     >**IMPORTANT**: In order to correctly send your profiles over, make sure you have
-    the following environment variables set: `[WHYLABS_ORG_ID, WHYLABS_API_KEY, WHYLABS_DEFAULT_DATASET_ID]`. You
-    can also follow the authentication instructions for the why.init() method at {_INIT_DOCS}.
+    the following environment variables set: `[WHYLABS_ORG_ID, WHYLABS_API_KEY, WHYLABSdefault_DATASET_ID]`. You
```
Woops. Can you tell I refactored something by doing a search/replace for `_default`...

Commit:
Addressed pr comments, fixed another type issue

Kind of surprised this worked the entire time. The ThreadQueueWrapper
never actually implemented close() or ABC. I guess that implies we don't
actually call it in the thread specific code, which makes sense since
the appropriate implementation is a no-op anyway.

richard-rogers commented on python/whylogs/api/logger/experimental/logger/actor/list_util.py:
```diff
@@ -1,17 +1,27 @@
 
-def get_like_items(items: List[Any]) -> Tuple[List[Any], Type, List[Any]]:
+
+def get_like_items(items: List[K]) -> Tuple[Optional[List[K]], Optional[Type[K]], List[K]]:
```
...verbal discussion...

I don't think it's possible to accurately annotate for both the input and output sides, so annotate how you like and add a comment explaining what's happening.

Commit:
Add comment to clarify types in a utility function.

Commit:
Add pyright to the rolling logger code

It's not much but its a start. We can add more as we go.

naddeoa committed:
Merge branch 'mainline' into pyright
